src/c-writer.cc

```diff
@@ -172,15 +172,22 @@ int GetShiftMask(Type type) {
 
 class CWriter {
  public:
-  CWriter(Stream* c_stream,
+  CWriter(std::vector<Stream*>&& c_streams,
           Stream* h_stream,
+          Stream* h_impl_stream,
           const char* header_name,
+          const char* header_impl_name,
           const WriteCOptions& options)
       : options_(options),
-        c_stream_(c_stream),
+        c_streams_(std::move(c_streams)),
         h_stream_(h_stream),
-        header_name_(header_name) {
+        h_impl_stream_(h_impl_stream),
+        header_name_(header_name),
+        header_impl_name_(header_impl_name) {
     module_prefix_ = MangleModuleName(options_.module_name);
+    if (c_streams_.empty()) {
+      c_streams_.push_back(h_stream);
```

Comment:
Why would `c_streams` be empty here?  Don't we pass in just one in the case of `-j1`?

Reply:
Updated in the latest commit.